### PR TITLE
Warn about routing with special chars /6

### LINKED ISSUE #27900: Add note about routing with special characters not working in .net core, 5+
Routing in .net (starting with core and at least up to 7) does not work for strings, nor is there a workaround possible.
The problem is that route parameters are half URL decoded by the middleware (Yes halve). So unless you know in advance what all possible routing values will be, routing WILL NOT and CAN NOT reliably work.

see https://github.com/dotnet/aspnetcore/issues/11544

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 8ee6ad2f-1667-8d5c-1f36-9d0511e3720f
* Version Independent ID: 7838ccec-c8e8-9baa-2b3e-220b2af5fc58
* Content: [Routing to controller actions in ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/mvc/controllers/routing?view=aspnetcore-7.0)
* Content Source: [aspnetcore/mvc/controllers/routing.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/controllers/routing.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

---
[Associated WorkItem - 60876](https://dev.azure.com/msft-skilling/Content/_workitems/edit/60876)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #27900

[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-8.0&branch=pr-en-us-28213#routing-with-special-characters)

## Public review  of the doc build:
- download the [HTML files](https://www.dropbox.com/sh/yy994551cje0kap/AAB0tRnm7gze2kYrSYPSFHIda?dl=0)
- unzip
- view HTML file in a browser.
- For links outside the doc, remove the `review.` from URL.

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@SchroterQuentin @ArthurHNL @DanielBryars @skovalyova @talshloman @celluj34 please review and suggest better wording. I'm hoping to make this very visible so folks can find it before spending a lot of time debugging. Hopefully those who hit this problem will click on the GitHub issue and provide feedback so we can get a good idea how common this problem is.

**Comment by thegrahamking** [comment]
I've only observed the behaviour with `/`, so can only comment that that is correct. I suggest adding wording regarding workarounds e.g. use query string params

File: aspnetcore/includes/routeSlash.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+```
+
+When `string id` contains the following encoded values, unexpected results might occur:
+
+| ASCII  | Encoded |
+| ----| ---------- |
+| `/` | `%2F`  |
+| `+` | `%2B`  |
+| ` ` | `+`  |
```
Comment (Tratcher) [inline aspnetcore/includes/routeSlash.md:26]: This is form encoding, not url encoding, it shouldn't affect the path.

File: aspnetcore/includes/routeSlash.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    return todoItem.Name;
+}
+```
+
+When `string id` contains the following encoded values, unexpected results might occur:
+
+| ASCII  | Encoded |
+| ----| ---------- |
+| `/` | `%2F`  |
+| `+` | `%2B`  |
```
Comment (Rick-Anderson) [inline aspnetcore/includes/routeSlash.md:25]:
```suggestion
```

**Review by Tratcher — APPROVED** [review]

**Comment by ArthurHNL** [comment]
@Rick-Anderson Sorry for my late reply, I got quite sick over the weekend. But this change looks good. Thanks for clearing this up in the documentation.

**Comment by ArthurHNL** [comment]
@Rick-Anderson Sorry for my late reply, I got quite sick over the weekend. But this change looks good. Thanks for clearing this up in the documentation.

**Comment by Mosch0512** [comment]
i feel like `:` should also be documentet here?

**Pull request merged**